src/gitHubCopilotForAzure.ts

```diff
@@ -0,0 +1,69 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { callWithTelemetryAndErrorHandling, IActionContext, openUrl } from "@microsoft/vscode-azext-utils";
+import { commands, Extension, ExtensionContext, extensions, window } from "vscode";
+import { localize } from "./utils/localize";
+
+const ghcpExtensionId = 'github.copilot';
+const ghcpChatExtensionId = 'github.copilot-chat';
+const ghcp4aExtensionId = 'ms-azuretools.vscode-azure-github-copilot';
+const ghcp4aLearnPage = 'https://aka.ms/GetGitHubCopilotForAzure';
+const dontShowKey = 'ghcp4a/dontShow';
+
+/**
+ * Conditionally show an install toast for the GitHub Copilot for Azure extension
+ */
+export function gitHubCopilotForAzureToast({ globalState }: ExtensionContext): void {
+    void callWithTelemetryAndErrorHandling('ghcp4aToast', async (context: IActionContext) => {
+        const arePrecursorExtensionsInstalled: boolean = isExtensionInstalled(ghcpExtensionId) && isExtensionInstalled(ghcpChatExtensionId);
+        if (!arePrecursorExtensionsInstalled || isExtensionInstalled(ghcp4aExtensionId)) {
+            return;
+        }
+
+        const dontShow: boolean = globalState.get<boolean>(dontShowKey, false);
+        if (dontShow) {
+            return;
+        }
+
+        const install = {
+            title: localize('install', 'Install'),
+            run: async () => {
+                context.telemetry.properties.install = 'true';
```

Comment:
Instead of three telemetry properties (`install`, `learnMore`, and `dontShowAgain`), I'd make this one property with three possible results.